components/kyma-environment-broker/internal/process/provisioning/ias_registration.go

```diff
@@ -31,49 +31,61 @@ func (s *IASRegistrationStep) Name() string {
 }
 
 func (s *IASRegistrationStep) Run(operation internal.ProvisioningOperation, log logrus.FieldLogger) (internal.ProvisioningOperation, time.Duration, error) {
-	spb := s.bundleBuilder.NewBundle(operation.InstanceID)
+	for spID := range ias.ServiceProviderInputs {
+		spb, err := s.bundleBuilder.NewBundle(operation.InstanceID, spID)
+		if err != nil {
+			msg := "failed to create new ServiceProvider Bundle"
+			log.Errorf("%s: %s", msg, err)
+			return s.operationManager.OperationFailed(operation, msg)
+		}
 
-	log.Info("Check if IAS ServiceProvider already exist")
-	err := spb.FetchServiceProviderData()
-	if err != nil {
-		return s.handleError(operation, err, log, "fetching IAS ServiceProvider data failed")
-	}
+		log.Infof("Check if IAS ServiceProvider %q already exist", spb.ServiceProviderName())
+		err = spb.FetchServiceProviderData()
+		if err != nil {
+			return s.handleError(operation, err, log, "fetching IAS ServiceProvider data failed")
+		}
 
-	if !spb.ServiceProviderExist() {
-		log.Info("Create IAS ServiceProvider")
-		err = spb.CreateServiceProvider()
+		if !spb.ServiceProviderExist() {
+			log.Infof("Create IAS ServiceProvider %q", spb.ServiceProviderName())
+			err = spb.CreateServiceProvider()
+			if err != nil {
+				return s.handleError(operation, err, log, "creating IAS ServiceProvider failed")
+			}
+		} else {
+			log.Infof("IAS ServiceProvider %q already registered", spb.ServiceProviderName())
+		}
+
+		log.Infof("Configure IAS ServiceProvider %q", spb.ServiceProviderName())
+		err = spb.ConfigureServiceProvider()
 		if err != nil {
-			return s.handleError(operation, err, log, "creating IAS ServiceProvider failed")
+			return s.handleError(operation, err, log, "configuring IAS ServiceProvider failed")
 		}
-	} else {
-		log.Infof("IAS ServiceProvider %q already registered", spb.ServiceProviderName())
-	}
 
-	log.Info("Configure IAS ServiceProvider")
-	err = spb.ConfigureServiceProvider()
-	if err != nil {
-		return s.handleError(operation, err, log, "configuring IAS ServiceProvider failed")
-	}
+		if spb.ServiceProviderType() == ias.OIDC {
```

Comment:
Does it make sense? first if then switch?
```go
spb.ServiceProviderType() == ias.OIDC {
(...)
  switch spID {
    case ias.SPGrafanaID:
    (...)
}
```
what otherr case could be in switch?